app/src/features/detail/components/PackageDetail.tsx

```diff
@@ -8,175 +8,234 @@ import {
-      >
-        <Tab label="Readme" />
-        <Tab label={`${versionCount} ${versionCount === 1 ? "Version" : "Versions"}`} />
-      </Tabs>
-      <Grid container spacing={4}>
-        {/* Main Content */}
-        <Grid item xs={12} md={8}>
-          <Box>
-            {activeTab === 0 && (
-              <Card
-                variant="outlined"
-                style={{
-                  marginBottom: "16px",
-                  borderRadius: "12px",
-                  boxShadow: "0 4px 12px rgba(0, 0, 0, 0.1)",
-                }}
-              >
-                <CardContent>
-                  <Typography
-                    variant="h6"
-                    gutterBottom
-                    style={{ fontWeight: "600", color: "#1976d2" }}
-                  >
-                    Readme
+        <CircularProgress />
+      </Box>
+    );
+  }
+
+  if (error) {
+    return (
+      <Box sx={{ margin: 4 }}>
+        <Alert severity="error">{error}</Alert>
+      </Box>
+    );
+  }
+
+  if (!data) {
+    return (
+      <Box sx={{ margin: 4 }}>
+        <Alert severity="info">No package information found.</Alert>
+      </Box>
+    );
+  }
+
+  const renderReadmeTab = () => (
+    <Card variant="outlined" className="card-dark">
+      <CardContent className="card-content">
+        {data.readme ? (
+          <div className="readme-content">
+            <ReactMarkdown>{data.readme}</ReactMarkdown>
+          </div>
+        ) : (
+          <Typography>No readme available for this package.</Typography>
+        )}
+      </CardContent>
+    </Card>
+  );
+
+  return (
+    <div className="package-detail-container">
+      <Container maxWidth="lg" className="package-detail-inner">
+        <div className="package-header">
+          <Typography variant="h4" gutterBottom className="package-title">
+            {data.name}@{data.version}
+          </Typography>
+          {data.description && (
+            <Typography
+              variant="body1"
+              className="package-description-text-header"
+            >
+              {data.description}
+            </Typography>
+          )}
+        </div>
+        <Tabs
+          value={activeTab}
+          onChange={handleTabChange}
+          indicatorColor="secondary"
+          textColor="inherit"
+          centered
+          className="package-tabs"
+          variant="scrollable"
+          scrollButtons="auto"
+        >
+          <Tab label="Readme" className="package-tab" />
+          <Tab label="Versions" className="package-tab" />
+          <Tab label="Dependencies" className="package-tab" />
+          <Tab label="Dependents" className="package-tab" />
+          <Tab label="Code" className="package-tab" />
+        </Tabs>
+
+        <Grid container spacing={4}>
+          {/* Main Content - Left Side (changes with tabs) */}
+          <Grid item xs={12} md={7} lg={8}>
+            {/* Readme Tab */}
+            {activeTab === 0 && renderReadmeTab()}
```

Comment:
nit: rather than conditional rendering based on `activeTab` index; maybe can put all these rendered components in a list and render using something like `tabList[activeTab]`